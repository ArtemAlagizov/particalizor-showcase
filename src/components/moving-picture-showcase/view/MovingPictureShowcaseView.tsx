import React from "react";
import { IMovingPictureShowcaseView } from "./IMovingPictureShowcaseView";
import { isNil } from "ramda";
import Dropzone from "react-dropzone";
import { MovingPicture } from "particalizor-3000";
import { Styled } from "../styled";
import "antd/dist/antd.css";
import cadillac from "../../../assets/testImage.png";
import SliderWithLabel from "../../../ui-components/slider-with-label/container/SliderWithLabel";
import DropdownWithLabel from "../../../ui-components/dropdown-with-label/container/DropdownWithLabel";
import SwitchWithLabel from "../../../ui-components/switch-with-label/container/SwitchWithLabel";

const MovingPictureShowcaseView: React.FC<IMovingPictureShowcaseView> = ({
  onParticleNumberChange,
  onTraceWidthChange,
  onVelocityChange,
  onLifeTimeChange,
  onDirectionChannelChange,
  onHueChannelChange,
  onHueReverseChange,
  onDirectionReverseChange,

  onDrop,

  particleNumberValue,
  particleTraceWidthValue,
  particleVelocityValue,
  particleLifeTimeValue,
  directionChannelValue,
  hueChannelValue,
  hueReversedValue,
  directionReversedValue,

  directionHueOptions,
  switchImageMode,
  switchButtonTitle,
  selectedImageMode,
  uploadedImage
}) => {
  return (
    <Styled.ShowcaseComponentContainer>
      <Styled.PropertiesContainer>
        <Styled.PropertiesSet>
          <SliderWithLabel
            propertyName={"particleNumber"}
            selectedValue={particleNumberValue}
            onValueChange={onParticleNumberChange}
            label={"particle number"}
            range={{ min: 200, max: 7000 }}
            step={200}
          />
          <SliderWithLabel
            propertyName={"particleTraceWidthValue"}
            selectedValue={particleTraceWidthValue}
            onValueChange={onTraceWidthChange}
            label={"particle trace width"}
            range={{ min: 1, max: 10 }}
          />
          <SliderWithLabel
            propertyName={"particleVelocityValue"}
            selectedValue={particleVelocityValue}
            onValueChange={onVelocityChange}
            label={"particle velocity"}
            range={{ min: 1, max: 10 }}
          />
          <SliderWithLabel
            propertyName={"particleLifeTimeValue"}
            selectedValue={particleLifeTimeValue}
            onValueChange={onLifeTimeChange}
            label={"particle life time"}
            range={{ min: 100, max: 1000 }}
            step={100}
          />
        </Styled.PropertiesSet>
        <Styled.PropertiesSet>
          <DropdownWithLabel
            propertyName={"directionChannel"}
            onValueChange={onDirectionChannelChange}
            selectedValue={directionChannelValue}
            label={"direction"}
            defaultValue={directionChannelValue}
            options={directionHueOptions}
          />
          <DropdownWithLabel
            propertyName={"hueChannel"}
            onValueChange={onHueChannelChange}
            selectedValue={hueChannelValue}
            label={"hue"}
            defaultValue={hueChannelValue}
            options={directionHueOptions}
          />
          <SwitchWithLabel
            propertyName={"hueReverse"}
            onValueChange={onHueReverseChange}
            label={"hue reversed"}
            defaultValue={hueReversedValue}
          />
          <SwitchWithLabel
            propertyName={"directionReverse"}
            onValueChange={onDirectionReverseChange}
            label={"direction reversed"}
            defaultValue={directionReversedValue}
          />
        </Styled.PropertiesSet>
      </Styled.PropertiesContainer>
      <Styled.ImageModeSwitch onClick={switchImageMode}>
        {switchButtonTitle}
      </Styled.ImageModeSwitch>
      {selectedImageMode === "uploaded-picture" && isNil(uploadedImage) && (
        <Styled.Images>
          <Styled.ImageContainer>
            <Styled.Dashed>
              <Dropzone onDrop={onDrop}>
                {({ getRootProps, getInputProps }) => (
                  <section>
                    <Styled.DropZone {...getRootProps()}>
                      <input {...getInputProps()} />
                      <Styled.DropImageText>
                        DRAG AND DROP AN IMAGE HERE, OR CLICK TO SELECT AN IMAGE
                      </Styled.DropImageText>
                    </Styled.DropZone>
                  </section>
                )}
              </Dropzone>
            </Styled.Dashed>
          </Styled.ImageContainer>
        </Styled.Images>
      )}
      {selectedImageMode === "uploaded-picture" && !isNil(uploadedImage) && (
        <Styled.Images>
          <Styled.ImageContainer>
            <Styled.Dashed>
              <Dropzone onDrop={onDrop}>
                {({ getRootProps, getInputProps }) => (
                  <section>
                    <div {...getRootProps()}>
                      <input {...getInputProps()} />
                      <Styled.ImageContainerImg
                        src={URL.createObjectURL(uploadedImage)}
                        alt={"dropped"}
                      />
                    </div>
                  </section>
                )}
              </Dropzone>
            </Styled.Dashed>
          </Styled.ImageContainer>
          <Styled.ImageContainer>
            <MovingPicture
              key={Date.now()}
              imageSource={URL.createObjectURL(uploadedImage)}
              particleTraceWidth={particleTraceWidthValue}
              particleVelocity={particleVelocityValue}
              particleNumber={particleNumberValue}
              particleLifeTime={particleLifeTimeValue}
              directionChannel={directionChannelValue}
              hueChannel={hueChannelValue}
              reverseHue={hueReversedValue}
              reverseDirection={directionReversedValue}
            />
          </Styled.ImageContainer>
        </Styled.Images>
      )}
      {selectedImageMode === "given-picture" && (
        <Styled.Images>
          <Styled.ImageContainer>
            <Styled.ImageContainerImg src={cadillac} alt={"original"} />
          </Styled.ImageContainer>
          <Styled.ImageContainer>
            <MovingPicture
              key={Date.now()}
              imageSource={cadillac}
              particleTraceWidth={particleTraceWidthValue}
              particleVelocity={particleVelocityValue}
              particleNumber={particleNumberValue}
              particleLifeTime={particleLifeTimeValue}
              directionChannel={directionChannelValue}
              hueChannel={hueChannelValue}
              reverseHue={hueReversedValue}
              reverseDirection={directionReversedValue}
            />
          </Styled.ImageContainer>
        </Styled.Images>
      )}
    </Styled.ShowcaseComponentContainer>
  );
};

export default MovingPictureShowcaseView;
