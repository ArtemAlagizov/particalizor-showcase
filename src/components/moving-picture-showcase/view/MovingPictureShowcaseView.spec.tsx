import React from "react";
import { shallow } from "enzyme";
import MovingPictureShowcaseView from "./MovingPictureShowcaseView";

describe("Given a MovingPictureShowcaseView", () => {
  describe("when rendering", () => {
    it("should match snapshot", () => {
      const component = shallow(
        <MovingPictureShowcaseView
          switchButtonTitle={""}
          uploadedImage={null}
          switchImageMode={jest.fn()}
          selectedImageMode={""}
          particleVelocityValue={1}
          particleTraceWidthValue={1}
          particleNumberValue={1}
          particleLifeTimeValue={1}
          onVelocityChange={jest.fn()}
          onTraceWidthChange={jest.fn()}
          onParticleNumberChange={jest.fn()}
          onLifeTimeChange={jest.fn()}
          onHueReverseChange={jest.fn()}
          onHueChannelChange={jest.fn()}
          onDirectionChannelChange={jest.fn()}
          hueReversedValue={false}
          hueChannelValue={"red"}
          directionReversedValue={false}
          directionHueOptions={[]}
          directionChannelValue={"hue"}
          onDirectionReverseChange={jest.fn()}
          onDrop={jest.fn()}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
