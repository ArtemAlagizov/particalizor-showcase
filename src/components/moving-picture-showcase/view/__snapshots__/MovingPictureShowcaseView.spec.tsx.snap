// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a MovingPictureShowcaseView when rendering should match snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <SliderWithLabel
        label="particle number"
        onValueChange={[MockFunction]}
        propertyName="particleNumber"
        range={
          Object {
            "max": 7000,
            "min": 200,
          }
        }
        selectedValue={1}
        step={200}
      />
      <SliderWithLabel
        label="particle trace width"
        onValueChange={[MockFunction]}
        propertyName="particleTraceWidthValue"
        range={
          Object {
            "max": 10,
            "min": 1,
          }
        }
        selectedValue={1}
      />
      <SliderWithLabel
        label="particle velocity"
        onValueChange={[MockFunction]}
        propertyName="particleVelocityValue"
        range={
          Object {
            "max": 10,
            "min": 1,
          }
        }
        selectedValue={1}
      />
      <SliderWithLabel
        label="particle life time"
        onValueChange={[MockFunction]}
        propertyName="particleLifeTimeValue"
        range={
          Object {
            "max": 1000,
            "min": 100,
          }
        }
        selectedValue={1}
        step={100}
      />
    </styled.div>
    <styled.div>
      <DropdownWithLabel
        defaultValue="hue"
        label="direction"
        onValueChange={[MockFunction]}
        options={Array []}
        propertyName="directionChannel"
        selectedValue="hue"
      />
      <DropdownWithLabel
        defaultValue="red"
        label="hue"
        onValueChange={[MockFunction]}
        options={Array []}
        propertyName="hueChannel"
        selectedValue="red"
      />
      <SwitchWithLabel
        defaultValue={false}
        label="hue reversed"
        onValueChange={[MockFunction]}
        propertyName="hueReverse"
      />
      <SwitchWithLabel
        defaultValue={false}
        label="direction reversed"
        onValueChange={[MockFunction]}
        propertyName="directionReverse"
      />
    </styled.div>
  </styled.div>
  <styled.div
    onClick={[MockFunction]}
  />
</styled.div>
`;
